### PR TITLE
feat: Add new pageOffset binding to Table Widget

### LINKED ISSUE #15013: [Feature]-[1842]:Add a new pageOffset binding in a table
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Today, when working with pagination in the queries, offset has to be written as {{(Table1.pageNo-1)*Table1.pageSize}}, this is hard for users to grok. We could simplify this by adding a new pageOffset which directly provides the value

### Why should this be worked on?

Easier server side pagination integration for our users

### PR DESCRIPTION
**Pull request by souma-ghosh**
## Description

Fixes #15013 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by sbalaji1192 — COMMENTED** [review]
Please write jest tests to validate the pageOffset binding

File: app/client/src/widgets/TableWidgetV2/widget/derived.js [hunk 1]
```diff
@@ -577,4 +577,14 @@ export default {
...
   },
   //
+  getPageOffset: (props, moment, _) => {
+    if (Number.isFinite(props.pageNo) && Number.isFinite(props.pageSize)) {
+      if (props.pageNo >= 0 && props.pageSize >= 0) {
+        /* Math.max fixes the value of (pageNo - 1) to a minimum of 0 as negative values are not valid */
+        return Math.max(props.pageNo - 1, 0) * props.pageSize;
+      }
+    }
+    return -1;
```
[the remark this answers is not in the document]
Comment (souma-ghosh) [reply app/client/src/widgets/TableWidgetV2/widget/derived.js:590]: Checked with Dilip. Default value is fine at -1.

**Comment by souma-ghosh** [comment]
@Aishwarya-U-R Some tests are failing here consistently. Could you please take a look?

**Review by sbalaji1192 — APPROVED** [review]

**Review by sbalaji1192 — DISMISSED** [review]

**Comment by laveena-en** [comment]
@souma-ghosh https://www.loom.com/share/39839e1979e844d7aab381243995498c On using the pageOffset binding in a query and run query on page load is off, an error is thrown on manually running the query. Can you please check?

Table1: SELECT * FROM users ORDER BY id LIMIT {{Table1.pageSize}} OFFSET {{(Table1.pageNo-1)*Table1.pageSize}};

Table2: SELECT * FROM users ORDER BY id LIMIT {{Table1.pageSize}} OFFSET {{(Table1.pageNo-1)*Table1.pageSize}};

**Comment by souma-ghosh** [comment]
@laveena-en Please go ahead with the QA. Latest DP is up

**Comment by chandannkumar** [comment]
Tested this PR and working as expected
- Tested pageOffset on Query as well as for server side pagination query
- Tested pageOffset on binding with widgets like Text widget

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**